language/move-lang/src/parser/byte_string.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::errors::*;
+use byteorder::{BigEndian, ByteOrder};
+use std::convert::TryFrom;
+
+use crate::parser::syntax::make_loc;
+
+/// Escape sequence.
+enum Sequence {
+    /// \x00
+    Hex(String),
+    ///u{7FFF}
+    Unicode(String),
```

Comment:
As I had mentioned in the community discussion (https://community.libra.org/t/string-literal-in-move/2830), we've had some internal discussions about Unicode and have so far concluded that we would like to hold off supporting it because it raises concerns that we haven't had time to consider carefully. Sorry if my comment about that was unclear.

Without any defined operations on strings in Move, Unicode is less scary than it would be otherwise, but even then, there could be implicit equality comparisons. We were thinking about a hypothetical scenario where a Move value is set to a Unicode bytestring and someone else has the same Unicode bytestring where a character has a different but equivalent encoding. If the correct behavior of the system depends on these values matching, there could be really subtle bugs that result. (I don't know if such a scenario actually exists now, so this is pretty hand-wavy.) In the longer term, we were wondering if it would be best to normalize all Unicode strings to a canonical representation so that string equality would behave better.

I suppose we could try to have that discussion now, but it will take time and will likely slow down acceptance of this PR. How much do you care about Unicode support?

Reply:
Removed Unicode support. But retained the ability to use Unicode using \u{...} notations.

Reply:
> But retained the ability to use Unicode using \u{...} notations.

I think the `\u{...}` notations are the Unicode support tools @bob-wilson is referring to?
But I could be wrong

Reply:
@bob-wilson do you have any examples you discussed with @vgao1996 for why even the `\u{...}` notations might be problematic?

Reply:
We discussed offline and we aren't going to want Unicode in the bytestring `b""` 

At some point we might have a string library, that parses with the syntax `""`. And in that version of strings, we would have this unicode escape sequences. 

With out the normalization of unicode sequences, too many funky edge cases exist that would feel awkward in `b""`